test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,159 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	_ "embed"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	e2e "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+)
+
+// LinuxVMFakeintakeSuite defines a test suite for the log agent interacting with a virtual machine and fake intake.
+type LinuxVMFakeintakeSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+	DevMode bool
+}
+
+//go:embed log-config/log-config.yaml
+var logConfig []byte
+
+// logsExampleStackDef returns the stack definition required for the log agent test suite.
+func logsExampleStackDef() *e2e.StackDefinition[e2e.FakeIntakeEnv] {
+
+	return e2e.FakeIntakeStackDef(
+		e2e.WithAgentParams(
+			agentparams.WithLogs(),
+			agentparams.WithIntegration("custom_logs.d", string(logConfig))))
+
+}
+
+// TestE2EVMFakeintakeSuite runs the E2E test suite for the log agent with a VM and fake intake.
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	s := &LinuxVMFakeintakeSuite{}
+	if _, devmode := os.LookupEnv("TESTS_E2E_DEVMODE"); devmode {
+		s.DevMode = true
+	}
+
+	e2e.Run(t, s, logsExampleStackDef())
+}
+
+func (s *LinuxVMFakeintakeSuite) BeforeTest(_, _ string) {
+	// Flush server and reset aggregators before the test is ran
+	s.cleanUp()
+}
+
+func (s *LinuxVMFakeintakeSuite) TearDownSuite() {
+	// Flush server and reset aggregators after the test is ran
+	s.cleanUp()
+}
+
+func (s *LinuxVMFakeintakeSuite) TestLinuxLogTailing() {
+	// Run test cases
+	s.Run("LogCollection", s.LogCollection)
+
+	s.Run("LogPermission", s.LogPermission)
+
+	s.Run("LogRotation", s.LogRotation)
+}
+
+func (s *LinuxVMFakeintakeSuite) LogCollection() {
+	t := s.T()
+	fakeintake := s.Env().Fakeintake
+
+	// Create a new log file
+	s.Env().VM.Execute("sudo touch /var/log/hello-world.log")
+
+	// Part 1: Ensure no logs are present in fakeintake
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		logs, err := fakeintake.FilterLogs("hello")
+		if !assert.NoError(c, err, "Unable to filter logs by the service 'hello'.") {
+			return
+		}
+		assert.Empty(c, logs, "Logs were found when none were expected.")
+
+		// If logs are found, print their content for debugging
+		if len(logs) != 0 {
+			cat, _ := s.Env().VM.ExecuteWithError("cat /var/log/hello-world.log")
+			t.Logf("Logs detected when none were expected: %v", cat)
+			assert.Empty(c, logs, "Logs were found when none were expected.")
+		}
```

Comment:
There is a some redundancy here.

It can be simplified with:
```suggestion
		if !assert.Empty(c, logs, "Logs were found when none were expected.") {
			// If logs are found, print their content for debugging
			cat, _ := s.Env().VM.ExecuteWithError("cat /var/log/hello-world.log")
			t.Logf("Logs detected when none were expected: %v", cat)
		}
```